x-pack/plugins/ml/public/application/util/url_state.ts

```diff
@@ -58,54 +58,87 @@ export function getUrlState(search: string): Dictionary<any> {
 //   different urlStates(e.g. `_a` / `_g`) don't overwrite each other.
 export const useUrlState = (accessor: string): UrlState => {
   const history = useHistory();
-  const { search } = useLocation();
+  const { search: locationSearchString } = useLocation();
+
+  // We maintain a local state of useLocation's search.
+  // This allows us to use the callback variant of setSearch()
+  // later on so we can make sure we always act on the
+  // latest url state.
+  const [searchString, setSearchString] = useState(locationSearchString);
+
+  useEffect(() => {
+    setSearchString(locationSearchString);
+  }, [locationSearchString]);
+
+  useEffect(() => {
+    // Only push to history if something related to the accessor of this
+    // url state instance is affected (e.g. a change in '_g' should not trigger
+    // a push in the '_a' instance).
+    if (
+      !isEqual(parseUrlState(locationSearchString)[accessor], parseUrlState(searchString)[accessor])
```

Comment:
looks like this effect should depend on two variables, you are missing `locationSearchString `

Reply:
I added more comments in 3500f11 to explain why `locationSearchString` isn't necessary for comparison in this case.